src/main/java/net/schmizz/sshj/common/SecurityUtils.java

```diff
@@ -275,17 +275,20 @@ public static synchronized void setSecurityProvider(String securityProvider) {
     }
 
     private static void register() {
-        if (!registrationDone) {
+        String noBCRegistrationAtRuntime = System.getProperty("bouncycastle.registration.runtime");
```

Comment:
Instead of doing it like this, let's make it just a configuration in the config object. This will allow a user of the library to choose how to configure it, either through an env variable or through his own config.
Now we would be forcing them to set an env variable.

Reply:
Thanks for the quick feedback! 👍  Please take another look.

Reply:
any updates? :)